solutions/src/main/java/ru/sberbank/school/task02/ReversedCurrencyCalc.java

```diff
@@ -0,0 +1,323 @@
+        }
+
+        boolean isBigger = beneficiary == Beneficiary.BANK ^ direction == ClientOperation.SELL;
+
+        /*if(price == null && Math.abs(delta - 0) > 1e-5) {
+
+            Position pos = checkBounds(lower, upper, amount.add(BigDecimal.valueOf(delta)),
+                                                     amount.subtract(BigDecimal.valueOf(delta)));
+
+            price = pickPrice(lower, upper, beneficiary, pos);
+        }*/
+
+        return isBigger ? Optional.of(prices.first().getPrice())
+                        : Optional.of(prices.last().getPrice());
+    }
+
+    /**
+     * Checks if amount is inside of one or more
+     * of its neighboring volumes' bounds.
+     * @return relative position of amount.
+     */
+    private Position checkBounds(Volume lowerLeft, Volume lowerRight,
+                                 Volume upperLeft, Volume upperRight,
+                                 BigDecimal amount) {
+        
+        Volume a = Volume.from(amount);
+
+        TreeSet<Volume> bounds = new TreeSet<>(this::compare);
+
+        bounds.add(lowerLeft);
+        bounds.add(lowerRight);
+        bounds.add(upperLeft);
+        bounds.add(upperRight);
+        bounds.add(a);
+
+        if (bounds.first().equals(a)) {
+            return Position.BOTTOM;
+        }
+        if (bounds.last().equals(a)) {
+            return Position.TOP;
+        }
+        if (lowerLeft.equals(bounds.lower(a)) && lowerRight.equals(bounds.higher(a))) {
+            return Position.LEFT;
+        }
+        if (upperLeft.equals(bounds.lower(a)) && upperRight.equals(bounds.higher(a))) {
+            return Position.RIGHT;
+        }
+        if (upperLeft.equals(bounds.lower(a)) && lowerRight.equals(bounds.higher(a))) {
+            return Position.INTERSECTION;
+        }
+
+        return Position.MIDDLE;
+    }
+
+    private Position checkBounds(Volume left, Volume right, BigDecimal amount) {
+
+        Volume a = Volume.from(amount);
+
+        if (compare(a, left) < 0) {
+            return Position.BOTTOM;
+        }
+        if (compare(a, right) > 0) {
+            return Position.TOP;
+        }
+
+        return Position.INTERSECTION;
+    }
+
+
+    /**
+     * Describes amount position relative to
+     * bounding volumes' values.
+     */
+    private enum Position {
+
+        /**
+         * Amount lies between two intersecting bounds.
+         */
+        INTERSECTION,
+
+        /**
+         * Amount is inside lower bounds.
+         */
+        LEFT,
+
+        /**
+         * Amount is inside upper bounds.
+         */
+        RIGHT,
+
+        /**
+         * Amount is between two non-intersecting bounds.
+         */
+        MIDDLE,
+
+        /**
+         * Amount is between upper bound and inf.
+         */
+        TOP,
+
+        /**
+         * Amount is between lower bound and zero.
+         */
+        BOTTOM
+    }
+
+    private int compare(Volume v1, Volume v2){
```

Comment:
WhitespaceAround: &#39;{&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1364944200>*